src/Microsoft.Health.Dicom.Api/Features/Routing/KnownRoutes.cs

```diff
@@ -21,29 +22,48 @@ public static class KnownRoutes
         private const string ExtendedQueryTagPathRouteSegment = "{" + KnownActionParameterNames.TagPath + "}";
 
         public const string StoreRoute = StudiesRouteSegment + "/{" + KnownActionParameterNames.StudyInstanceUid + "?}";
+        public const string VersionedStoreRoute = VersionSegment + "/" + StoreRoute;
```

Comment:
Gosh, thank you for adding all of these. But this also look like a big pain to write lol. I tried something that I think should work here; we should extend the `RouteAttribute` class to automatically prepend the version. Ideally we'd define a new `Attribute` that extends `RouteAttribute` and annotate our controller actions with this new attribute too.

Eg.

```csharp
[Route("Foo/Bar")]
[VersionedRoute("Foo/Bar")]
public async Task<IActionResult> GetResponse()
{
    // ...
}
```

Furthermore, it'd be even cooler if we could combine them. Perhaps there is a template syntax that could make the verison optional which we could control via an argument `[VersionedRoute("Foo/Bar", optionalVersion = true)]`

Reply:
Great idea - I've added a VersionedRoute attribute.
I didn't add the optionalVersion param though - I think the idea is to get all customers to move off of the non-versioned route & then get rid of it. So routes without versions should go away soon-ish